HLshell/mycmd_1.py

```diff
@@ -334,57 +326,175 @@ def get_names(self): ## Don't change the name
 ## Laptop
 class clientShell(shell):
     DEFAULT_TIMEOUT = 6
-    def __init__(self):
+    def __init__(self, ip):
         super().__init__()
         self.intro = '####### Interactive Client Shell #######'
         self.prompt = 'cmd> '
         self.doRemote = True
+        self.server_ip = ip
+        self.is_connected = False 
+        self.is_connecting = False
+        self.waiting_for_cmd = False
+        self.debug_mode = False
 
     def preloop(self):
-        pass
-        #print('Open Client Shell')
+        '''Automatically connects and waits for commands'''
+       
+        self.do_start('')
+
+       
+    
+    @parser()
+    def do_start(self, *par):
+        '''start [IP] 
+        
+        Start automatically connecting and receiving commands from server'''
+        self.debug_mode = False
+        if(len(par) > 0):
+            self.server_ip = par
+
+        # I probably should've switched preloop and start
+        pressed_key = '_'
+        EXIT = 'q'
+        self.is_connecting = False
+
+        while(not self.debug_mode):
+            connect_thread = threading.Thread(target=self.do_connect, args=[self.server_ip])
```

Comment:
If we check connection in background and send the command, does the port collision occur?
I set the default port is 8001. If the collision happens, port 800X is available.

Reply:
I made sure to only try connecting if the client isn't already connecting, so there should only be one connect request at a time.